### PR TITLE
Set ineligible issues as closed (also some refactoring)

### PR DESCRIPTION
**Pull request by shanear**
connects #9875

A follow up PR to migrate all the old data will be coming. For now the code works for both closed and "open" ineligible issues.

### CODE REVIEW

File: app/models/request_issue.rb [hunk 1]
```diff
@@ -2,17 +2,6 @@
 
 # rubocop:disable Metrics/ClassLength
 class RequestIssue < ApplicationRecord
-  # TODO: remove this eventually, used to protect caching from screwing up removed columns
```
Comment (shanear) [inline app/models/request_issue.rb]: figured now was as good of time as ever to get rid of this.

File: app/models/request_issue.rb [hunk 2]
```diff
@@ -387,30 +378,35 @@ def legacy_issue_opted_in?
     eligible? && vacols_id && vacols_sequence_id
   end
 
-  def remove!
-    update!(closed_at: Time.zone.now, closed_status: :removed)
+  def close!(status)
```
Comment (shanear) [inline app/models/request_issue.rb]: factors a lot of the close logic into one place

File: spec/models/request_issue_spec.rb [hunk 3]
```diff
@@ -71,7 +72,8 @@
       vacols_id: vacols_id,
       vacols_sequence_id: vacols_sequence_id,
       closed_at: closed_at,
-      closed_status: closed_status
+      closed_status: closed_status,
+      ineligible_reason: ineligible_reason
```
Comment (shanear) [inline spec/models/request_issue_spec.rb]: decided to speckle some validations into the feature tests for extra coverage.

File: app/models/request_issue.rb (hunk 2 above)
Comment (leikkisa) [reply app/models/request_issue.rb]: Nice

**Pull request merged**